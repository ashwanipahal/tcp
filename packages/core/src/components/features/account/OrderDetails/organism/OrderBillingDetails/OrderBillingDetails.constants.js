const cardIconMapping = {
  DISC: 'disc-small',
  'MASTER CARD': 'mc-small',
  AMEX: 'amex-small',
  VISA: 'visa-small',
  'GIFT CARD': 'gift-card-small',
  'PLACE CARD': 'place-card-small',
  VENMO: 'venmo-blue-acceptance-mark',
  PAYPAL: 'paypal-icon',
};

export default cardIconMapping;
