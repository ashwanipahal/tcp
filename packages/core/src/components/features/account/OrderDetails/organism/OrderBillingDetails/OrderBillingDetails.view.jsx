import React from 'react';
import PropTypes from 'prop-types';
import { BodyCopy } from '@tcp/core/src/components/common/atoms';
import withStyles from '@tcp/core/src/components/common/hoc/withStyles';
import styles from './styles/OrderBillingDetails.style';

/**
 * This function component use for return the EarnPoints
 * can be passed in the component.
 * @param waysToEarn - waysToEarn object used for showing extra points details
 */

const OrderBillingDetails = ({ className }) => {
  /**
   * @function return  Used to render the JSX of the component
   * @param    {[Void]} function does not accept anything.
   * @return   {[Object]} JSX of the component
   */

  return (
    <BodyCopy component="div" className={className}>
      <BodyCopy fontSize="fs14" fontWeight="extrabold" fontFamily="secondary">
        Billing
      </BodyCopy>
    </BodyCopy>
  );
};
OrderBillingDetails.propTypes = {
  className: PropTypes.string,
};

OrderBillingDetails.defaultProps = {
  className: '',
};

export default withStyles(OrderBillingDetails, styles);
export { OrderBillingDetails as OrderBillingDetailsVanilla };
