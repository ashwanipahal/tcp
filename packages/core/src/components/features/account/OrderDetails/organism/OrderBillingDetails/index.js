export { default } from './OrderBillingDetails.view';
