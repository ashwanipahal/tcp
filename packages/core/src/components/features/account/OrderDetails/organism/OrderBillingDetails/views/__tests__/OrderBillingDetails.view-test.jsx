import React from 'react';
import { shallow } from 'enzyme';
import OrderBillingDetails from '../OrderBillingDetails.view';

describe('Order Billing Details component', () => {
  it('should renders correctly', () => {
    const props = {
      ordersLabels: {},
      className: '',
    };
    const component = shallow(<OrderBillingDetails {...props} />);
    expect(component).toMatchSnapshot();
  });
});
