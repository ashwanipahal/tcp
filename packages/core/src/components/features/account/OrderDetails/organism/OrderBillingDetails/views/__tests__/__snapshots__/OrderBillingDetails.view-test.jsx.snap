// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order Billing Details component should renders correctly 1`] = `
<StyledComponent
  className=""
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-VigVT",
        "isStatic": false,
        "rules": Array [
          "
  ",
          "
  .card-details {
    display: flex;
    align-items: center;
    margin: ",
          [Function],
          " 0;
  }
",
          ";
",
        ],
      },
      "displayName": "Styled(OrderBillingDetails)",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-VigVT",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  orderDetailsData={
    Object {
      "checkout": Object {
        "billing": Object {
          "card": Object {
            "VISA": "***************1211",
          },
        },
      },
    }
  }
  ordersLabels={Object {}}
/>
`;
